src/colorizer/canvas/elements/footer.ts

```diff
@@ -0,0 +1,110 @@
+import { Vector2 } from "three";
+
+import {
+  BaseRenderParams,
+  ContainerStyle,
+  defaultContainerStyle,
+  defaultFontStyle,
+  EMPTY_RENDER_INFO,
+  FontStyle,
+  RenderInfo,
+} from "../types";
+import { getInsetBoxRenderer, InsetBoxStyle } from "./insetBox";
+import { getLegendRenderer, LegendParams, LegendStyle } from "./legend";
+import { getScaleBarRenderer, ScaleBarParams, ScaleBarStyle } from "./scaleBar";
+import { getTimestampRenderer, TimestampParams, TimestampStyle } from "./timestamp";
+
+export type FooterStyle = ContainerStyle & FontStyle;
+
+export type FooterParams = BaseRenderParams & {
+  visible: boolean;
+
+  timestamp: TimestampParams;
+  timestampStyle: TimestampStyle;
+  scaleBar: ScaleBarParams;
+  scaleBarStyle: ScaleBarStyle;
+  insetBoxStyle: InsetBoxStyle;
+  legend: LegendParams;
+  legendStyle: LegendStyle;
+};
+
+export const defaultFooterStyle: FooterStyle = {
+  ...defaultFontStyle,
+  ...defaultContainerStyle,
+  paddingPx: new Vector2(10, 10),
+};
+
+/**
+ * Returns the render information for the footer, which contains the legend and an inset box with
+ * the timestamp and scale bar inside. If the footer is not visible, the inset box is rendered in
+ * the bottom right corner of the viewport.
```

Comment:
When the footer is not visible, the inset box is rendered in the bottom right corner: 
![image](https://github.com/user-attachments/assets/e59eb1d5-8a29-4a34-8e7c-3b87d1111ed8)

When the footer is visible:

![Small-volume-015 (6)](https://github.com/user-attachments/assets/54d2ad5c-b150-47b2-983c-c0a466819a1a)